.github/workflows/building.yaml

```diff
@@ -52,3 +52,13 @@ jobs:
           name: econnect-alarm-${{ env.PKG_VERSION }}.tar.gz
           path: dist/econnect_alarm-${{ env.PKG_VERSION }}.tar.gz
           if-no-files-found: error
+
+      - name: ZIP Component Dir
+        run: |
+          cd ${{ github.workspace }}/custom_components/econnect_alarm
+          zip econnect_alarm.zip -r ./
+
+      - name: Upload zip to release
```

Comment:
This step is good if we manage to trigger it when the event is `publish`. I think you can find more details in the [HACS repository](https://github.com/hacs/integration/blob/main/.github/workflows/publish.yml#L61)!

Other than that, I would recommend to still publish the artifact, so that we can attach it to the run that generated it. It's useful for debug purposes. This is an example from this file: https://github.com/palazzem/ha-econnect-alarm/blob/68e39bfd05f88ce8c6e2d10f134c68f80d73d41c/.github/workflows/building.yaml#L49-L54

Reply:
So far this is just a zipped version of the content of "custom_components", so I'll leave it this way and eventually leverage the "download source" if needed.

Reply:
Sure thing!